mypy/server/astdiff.py

```diff
@@ -83,7 +83,10 @@ def is_similar_node_shallow(n: SymbolTableNode, m: SymbolTableNode) -> bool:
                 nn.is_newtype == mn.is_newtype and
                 is_same_mro(nn.mro, mn.mro))
     if isinstance(n.node, Var) and isinstance(m.node, Var):
-        return is_identical_type(n.node.type, m.node.type)
+        if n.node.type is None and m.node.type is None:
```

Comment:
> But if you're now no longer sure that it should return ``True`` if both are ``None``, then @JukkaL must clarify.

Actually, we discussed this with Jukka in person and he said that it is OK. If there are no more questions then this could be merged, I leave this up to you and/or Jukka.